src/main/scala/example/models/Book.scala

```diff
@@ -5,7 +5,10 @@ import scala.util.{Try, Success, Failure}
 import example.models.errors.{TransformError}
 import example.models._
 
-case class Book(isbn: String)
+case class Book(isbn: String) {
+  def linkToGoodreads: String = s"https://www.goodreads.com/book/isbn/$isbn"
```

Comment:
To me this feels like we're tightly coupling the `Book` case class to always having a goodreads link. Could we do something with `Book` similar to what you're doing with your `Error` trait so that it's more extensible?

Reply:
I'm confused as to why it matters if a book has a goodreads link or not. Why should it not? also, aside from making another function in a different file, no, I don't know how to apply what i did with the error trait to this problem.

Reply:
`Book` is very generic while `linkToGoodReads` is very specific. The current setup couples a generic object with a specific implementation of a link. What if you wanted to offer links to multiple different book websites? 

In `Error` you have a generic trait that gets extended to more specific implementation of errors that extend the base `Error` trait. That is one way of making you `Book` object more extensible but not the only way.

Reply:
I understand. The format of the string being tied to the generic book class, got it. Well I don't know exactly what I would do about that.

Reply:
You could do something very similar to what you’re doing with Error. Have a generic Book trait with unimplemented function signatures. Create case classes that extend that trait (exactly what you’re doing in Error) with the case classes being more specific, such as ‘GoodReadsBook’

Reply:
Then ‘GoodReadsBook’ houses the specific implementation details of ‘def getLink()’ and ‘apply()’. You’d then leave your project open to easily add more book case classes that can be built from sources outside of goodreads.

Reply:
```
trait Book {
  def link: String
}

case class GoodReadsBook(isbn: String) extends Book {
  def link: String = ???
  def apply
}

object GoodReadsBook {
  def apply(vol: Volume): Either[TransformError, Book] =
    Some(vol)
      .map(_.volumeInfo)
      .flatMap(_.industryIdentifiers)
      .flatMap(_.lift(0))
      .map(_.identifier)
      .map(Book.apply)
      .toRight(TransformError("No ISBN in response"))
}

case class GoogleBooksBook(isbn: String) extends Book {
  def link: String = ???
}

object GoodReadsBook {
  def apply(): Either[TransformError, Book] = ???
}
```

Reply:
That payload is from google, so the order of this would be flipped around, but i see what you are saying.